src/Sil/Bundle/ProductBundle/Controller/AttributeTypeController.php

```diff
@@ -0,0 +1,230 @@
+                            'attributeTypeId' => '%item%.id',
+                        ],
+                    ],
+                ],
+                'pagination'  => $pagination,
+            ],
+        ]);
+    }
+
+    /**
+     * show route.
+     *
+     * @param string $attributeTypeId
+     *
+     * @return Response
+     */
+    public function showAction(string $attributeTypeId): Response
+    {
+        $attributeType = $this->findAttributeTypeOr404($attributeTypeId);
+
+        $this->breadcrumbBuilder
+            ->addItemWithRoute(
+                $this->get('translator')->trans('sil.product.attribute_type.list.page_header.title'),
+                'sil_product_attribute_type_list'
+            )
+            ->addItemWithRoute(
+                $attributeType->getName(),
+                'sil_product_attribute_type_show',
+                ['attributeTypeId' => $attributeTypeId]
+            )
+        ;
+
+        $attributeTypeGeneralForm = $this->createForm(GeneralType::class, $attributeType, [
+            'action' => $this->generateUrl('sil_product_attribute_type_edit', [
+                'attributeTypeId' => $attributeType->getId(),
+                'formName'        => 'general',
+            ]),
+        ]);
+
+        return $this->render('@SilProduct/AttributeType/show.html.twig', [
+            'attributeType' => $attributeType,
+            'form'          => [
+                'general' => $attributeTypeGeneralForm->createView(),
+            ],
+        ]);
+    }
+
+    /**
+     * create route.
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function createAction(Request $request): Response
+    {
+        $form = $this->createForm(CreateType::class);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $attributeType = $form->getData();
+
+            $this->attributeTypeRepository->add($attributeType);
+
+            $this->addFlash('success', $this->get('translator')->trans('sil.product.flashes.create.success'));
+
+            return $this->redirectToRoute('sil_product_attribute_type_show', [
+                'attributeTypeId' => $attributeType->getId(),
+            ]);
+        }
+
+        return $this->render('@SilProduct/AttributeType/create.html.twig', [
+            'form' => $form->createView(),
+        ]);
+    }
+
+    public function editAction(Request $request, string $attributeTypeId): Response
+    {
+        $attributeType = $this->findAttributeTypeOr404($attributeTypeId);
+
+        $form = $this->createForm(GeneralType::class, $attributeType, [
+            'action' => $this->generateUrl('sil_product_attribute_type_edit', [
+                'attributeTypeId' => $attributeType->getId(),
+            ]),
+        ]);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $attribute = $form->getData();
```

Comment:
**$attribute** is not used anywhere. Maybe a typo ?

Reply:
Yes this is under fix, the var would be named `$attributeType`.